athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaRecordHandler.java

```diff
@@ -0,0 +1,318 @@
+            throws IOException {
+        logger.info("readWithConstraint: schema[{}] tableName[{}]", recordsRequest.getSchema(), recordsRequest.getTableName());
+
+        Schema schemaName = recordsRequest.getSchema();
+
+        Split split = recordsRequest.getSplit();
+        String id = split.getProperty("query_id").replace("-","");
+        String fileLocation = split.getProperty("fileLocation");
+        String exportBucket = split.getProperty("exportBucket");
+
+        //get column name and type from the Schema
+        HashMap<String, Types.MinorType> mapOfNamesAndTypes = new HashMap<>();
+        for(Field field : schemaName.getFields()){
+            Types.MinorType minorTypeForArrowType = Types.getMinorTypeForArrowType(field.getType());
+            mapOfNamesAndTypes.put(field.getName(), minorTypeForArrowType);
+        }
+
+        // creating a RowContext class to hold the column name and value.
+        final RowContext rowContext = new RowContext(id);
+
+        //Generating the RowWriter and Extractor
+        GeneratedRowWriter.RowWriterBuilder builder = GeneratedRowWriter.newBuilder(recordsRequest.getConstraints());
+        for (Field next : recordsRequest.getSchema().getFields()) {
+            Extractor extractor = makeExtractor(next, rowContext, mapOfNamesAndTypes);
+            builder.withExtractor(next.getName(), extractor);
+        }
+        GeneratedRowWriter rowWriter = builder.build();
+
+        /*
+         Using S3 Select to read the S3 Parquet files generated in the split
+         */
+
+        //List all the parquet files generated in the split
+        ObjectListing objectListing = amazonS3.listObjects(new ListObjectsRequest().withBucketName(exportBucket).withPrefix(id));
+        for (S3ObjectSummary objectSummary : objectListing.getObjectSummaries()) {
+            if(objectSummary.getSize() > 0){
+                logger.info("Object Key is : " + objectSummary.getKey() + " Object Summary is: " + objectSummary.getSize());
+
+                //Creating the read Request
+                SelectObjectContentRequest request = generateBaseParquetRequest(exportBucket, objectSummary.getKey());
+                try (SelectObjectContentResult result = amazonS3.selectObjectContent(request)) {
+                    InputStream resultInputStream = result.getPayload().getRecordsInputStream();
+                    JSONParser jsonParser = new JSONParser();
+
+                    BufferedReader streamReader = new BufferedReader(new InputStreamReader(resultInputStream, "UTF-8"));
+                    JSONArray jsonArray = new JSONArray();
+                    String inputStr;
+
+                    //map to contain the column name and value pair.
+                    HashMap<String, Object> map = new HashMap<>();
+
+                    while ((inputStr = streamReader.readLine()) != null) {
+                        JSONObject data = (JSONObject) jsonParser.parse(inputStr);
```

Comment:
I thought we were getting parquet files from vertica?

Reply:
The AWS S3 SDK provides a [InputSerialization](https://docs.aws.amazon.com/AmazonS3/latest/API/API_InputSerialization.html) class for Parquet format, but an [OutputSerialization](https://docs.aws.amazon.com/AmazonS3/latest/API/API_OutputSerialization.html) only for CSV and JSON.  I am trying to use S3 Select to read the parquet file but have to serialize the result as a JSON to read it. Let's chat offline if we need to make changes to this.

Reply:
I didn't realize we were planning to use S3 Select. Let's connect offline about this. I'm pretty sure this will be slower than if we did it natively ourselves since we will always being doing a select *.